python/lsst/source/injection/inject_engine.py

```diff
@@ -533,6 +542,15 @@ def inject_galsim_objects_into_exposure(
             )
             bitvalue = exposure.mask.getPlaneBitMask(mask_plane_name)
             exposure[common_box].mask.array |= bitvalue
```

Comment:
As for the why, it was an initial suggestion from @TallJimbo. We should probably circularize this, rather than leave it as a box. Perhaps we can also get away with 3x3 instead. Essentially, this mask just needs to be big enough to always be able to identify the core/peak of the injected source. For larger injected sources, the detected centroid may drift from its known input.